kviewmodel/src/commonMain/kotlin/com/adeo/kviewmodel/KViewModel.kt

```diff
@@ -1,17 +1,29 @@
 package com.adeo.kviewmodel
 
+import kotlinx.coroutines.CoroutineExceptionHandler
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.SupervisorJob
 import kotlinx.coroutines.cancel
+import kotlin.coroutines.CoroutineContext
+import kotlin.native.concurrent.ThreadLocal
 
 public abstract class KViewModel {
 
+    private val mainCoroutineExceptionHandler: CoroutineExceptionHandler? by lazy {
+        getCoroutineExceptionHandler() ?: sharedExceptionHandler
+    }
     private val coroutineTags = hashMapOf<String, CoroutineScope>()
+    private val mainCoroutineContext = (SupervisorJob() + Dispatchers.Main.immediate).run {
+        val exceptionHandler = mainCoroutineExceptionHandler ?: return@run this
+        this + exceptionHandler
+    }
 
     public val viewModelScope: CoroutineScope
         get() = coroutineTags[MAIN_JOB_KEY] ?: launchNewScope()
 
+    protected open fun getCoroutineExceptionHandler(): CoroutineExceptionHandler? = sharedExceptionHandler
```

Comment:
Maybe we can make a more kotlin-friendly API, remove `getCoroutineExceptionHandler` method and make `mainCoroutineExceptionHandler` protected open?

```kotlin
protected open val mainCoroutineExceptionHandler: CoroutineExceptionHandler?
    get() = sharedExceptionHandler
```

Reply:
Yes, of course! 
I would like to do that, but it might not be safe in some cases, unfortunately. 
If derived class (class that extend other class) override that variable like this:
`override val mainCoroutineExceptionHandler: CoroutineExceptionHandler? = myCustomExceptionHandler`
it will return `null` inside `mainCoroutineContext`. I have already tried it :(

So I decided to write less kotlin-friendly, but more safety code

Reply:
Oh, I see. You're right.